packages/patternfly-4/src/content/design-guidelines/usage-and-behavior/login/design.md

```diff
@@ -0,0 +1,173 @@
+
+#### Client-side Errors
+Error messages are presented before the form is submitted. Client-side validation can be performed as soon as a field loses focus, and allows you to use inline validation to alert users of errors or issues immediately. In most cases, client-side validations are performed as inline validations for each field. Some common examples are shown below.
+
+##### Empty Field(s)
+
+
+![Empty-fields](empty-fields.png "empty-fields")
+
+An error message should be displayed below any required field that is empty once the action button (Log In/Next) has been clicked. Once the field has been filled, the error message should disappear. The following error messages are suggested for use.
+
+| Situation     | Error Message
+| ------------- |-------------|
+| Username field is empty | Enter your email, phone number, or username|
+| Password field is empty      | Enter your password |
+
+
+##### Invalid Characters
+
+![Invalid-characters](invalid-characters.png "invalid-characters")
+
+An error message should be displayed when there is an invalid character in the username. Once the field has been modified, the error message should disappear. The following error messages are suggested for use.
+
+| Situation     | Error Message
+| ------------- |--------------|
+| Invalid email address or phone number   | Enter a valid email address  |
+| Invalid custom username                 | Invalid characters           |
+| Invalid character in an email address/phone number | Enter a valid email address or phone number |
+
+
+
+##### Caps lock is on
+
+![Multi-factor-login-validation-options](caps-lock-on.png "Multi-factor-login-validation-options")
+
+
+A warning message should be displayed when caps lock is on so users avoid making a mistake when typing a case-sensitive password. The warning message should be displayed in a popover when a user clicks into the input field. Once caps lock has been turned off, the warning message should disappear. The following warning message is suggested for use.
+
+
+| Situation     | Error Message|
+| ------------- |-------------|
+| Caps lock is on   | Caps lock is on  |
+
+
+#### Server-side Errors
+When users submit a form, the entire page is reloaded if there are validation errors. At the same time, the password field should be cleared to provide additional security. Error messages can appear as inline errors for each field or as an inline error displayed below the application name and above the input fields. If there are multiple messages, each message appears on its own line; spacing between fields should be enough so that the form doesn’t jump to fit the messages.
+
+
+
+##### Account doesn’t exist/password is wrong
+
+
+![Invalid-characters](account-doesn't-exist:password-is-wrong.png "invalid-characters")
```

Comment:
Looks like there is an image here that is not displaying.

Reply:
Fixed filename for image.